src/components/media/controls/controls.tsx

```diff
@@ -0,0 +1,274 @@
+    volumeUp: { symbol: 'volume_up', label: 'mute' },
+    volumeOff: { symbol: 'volume_off', label: 'unmute' },
+  }
+  return (
+    <IconButton
+      onClick={props.onClick}
+      aria-label={`click to ${controls[props.action].label}`}
+      className={props.className}
+      data-cy={`${props.action}-button`}
+    >
+      <Icon color="primary">
+        <span className="material-symbols-rounded">
+          {controls[props.action].symbol}
+        </span>
+      </Icon>
+    </IconButton>
+  )
+}
+
+interface MediaControls {
+  mediaElement: HTMLMediaElement
+}
+
+export function ProgressSlider(props: MediaControls) {
+  const formattedElapsedTime = formatDurationTime(
+    props.mediaElement.currentTime
+  )
+
+  function handleTimeChange(event: Event, newValue: number | number[]) {
+    props.mediaElement.currentTime = newValue as number
+  }
+
+  function renderBufferedProgressBar() {
+    const buffered = props.mediaElement.buffered
+    const percentMultiple = 100
+    let secondsBuffered = 0
+
+    // // This loop calculates the total seconds buffered up to the current playback time by iterating through the buffered time ranges in reverse order. It finds the most recent buffered range that comes before or includes the current time and uses its end point to calculate the total buffered percentage. This ensures we capture the relevant buffered duration relative to the current playback position, even in scenarios where buffering is discontinuous or occurs in segments.
+    for (let i = 0; i < buffered.length; i++) {
+      const reverseIndex = buffered.length - 1 - i
+      const rangeStart = buffered.start(reverseIndex)
+      const rangeEnd = buffered.end(reverseIndex)
+
+      if (rangeStart < props.mediaElement.currentTime) {
+        secondsBuffered = rangeEnd
+        break
+      }
+    }
+
+    const bufferedPercent =
+      (secondsBuffered / props.mediaElement.duration) * percentMultiple
+
+    return (
+      <LinearProgress
+        className="rustic-progress-buffered"
+        variant="determinate"
+        value={bufferedPercent}
+      />
+    )
+  }
+
+  return (
+    <Box className="rustic-progress">
+      {renderBufferedProgressBar()}
+      <Slider
+        className="rustic-progress-slider"
+        data-cy="progress-slider"
+        size="small"
+        aria-label="Time"
+        aria-valuetext={`Current time: ${formattedElapsedTime}`}
+        max={props.mediaElement.duration}
+        value={props.mediaElement.currentTime}
+        onChange={handleTimeChange}
+        valueLabelDisplay="auto"
+        valueLabelFormat={(value) => formatDurationTime(value)}
+        color="secondary"
+      />
+    </Box>
+  )
+}
+
+export function VolumeSettings(props: MediaControls) {
+  const [volumeFraction, setVolumeFraction] = useState(1)
+
+  const isMuted = props.mediaElement.muted
+  const action = isMuted ? 'volumeOff' : 'volumeUp'
+  const percentMultiple = 100
```

Comment:
Maybe we could define this at the top of the file, I've seen it used in 2 different functions.